crates/ruff_linter/src/rules/pylint/snapshots/ruff_linter__rules__pylint__tests__PLE2510_invalid_characters.py.snap

```diff
@@ -46,7 +46,7 @@ invalid_characters.py:55:21: PLE2510 [*] Invalid unescaped character backspace,
    |
 53 | zwsp_after_multicharacter_grapheme_cluster = f"ಫ್ರಾನ್ಸಿಸ್ಕೊ ​​"
 54 | 
-55 | nested_fstrings = f'␈{f'{f'␛'}'}'
+55 | nested_fstrings = f'␈{f'{f'␛'}'}'
```

Comment:
Are these bugs in anstream? Or is there something strange going on with the invalid characters here?

Reply:
Not sure. Could you take a look at the file in a hex viewer? Maybe the file ending changed or it's something else but I'm surprised that there are changed snapshots.

Anoter possibility is that your `cargo insta` version is too old/new

Reply:
On `main`, this line contains a [substitute control character](https://en.wikipedia.org/wiki/Substitute_character) (`␚`, `^Z`; hex `\x1a`), which is not present on this branch:

![image](https://github.com/user-attachments/assets/272dba32-427b-43e4-901c-006fb9d265a3)

This character gets stripped by `anstream`, as demonstrated by this snippet:
```rs
let styled_text = "foo\x1abar";
let plain_str = anstream::adapter::strip_str(&styled_text).to_string();
assert_eq!(plain_str, "foobar");
```
Possibly because they use `0x1a` to switch to another mode here: https://github.com/rust-cli/anstyle/blob/5628f47a37e059d468c9ea2eab5ce69950e794e9/crates/anstyle-parse/src/state/codegen.rs#L59 — I haven't investigated further.

I don't know if this is a bug in anstream or expected behavior, but I think this reveals a problem with this strip-ansi-escape-sequences approach. Some of these snapshot files could contain actual (non-escaped) `1b 5b` escape sequences that should not be stripped.

`colored` offers a way to [disable colors at runtime](https://docs.rs/colored/latest/colored/control/struct.SHOULD_COLORIZE.html#method.set_override). Would that be an alternative approach?

Reply:
> colored offers a way to [disable colors at runtime](https://docs.rs/colored/latest/colored/control/struct.SHOULD_COLORIZE.html#method.set_override). Would that be an alternative approach?

One issue with this approach is that the colorize "leaks" into other tests and it would disable colorized output in mdtests depending on the test-execution order.

Reply:
> I don't know if this is a bug in anstream or expected behavior, but I think this reveals a problem with this strip-ansi-escape-sequences approach.

I do expect that this is intended because the strip stream strips any non-printable characters. I don't think this is necessarily desired in our case because we want to test and catch unprintable characters in our test outputs.

Reply:
> `colored` offers a way to [disable colors at runtime](https://docs.rs/colored/latest/colored/control/struct.SHOULD_COLORIZE.html#method.set_override). Would that be an alternative approach?

This is how we disable colours in the tests for mdtest itself, FWIW:

https://github.com/astral-sh/ruff/blob/2fe203292aa17a34dc878684ad2f1003504ea7f5/crates/red_knot_test/src/matcher.rs#L367